Software/scan_key/main.py

```diff
@@ -42,16 +43,40 @@
         while (ID == holdID and holdID != None and holdID[0:5] == "02350"):
             try:
                 author = machineAuth(ID, cursor)
-                if(author == True):
+                if (author == True):
                     break
-                # if the PIN was incorrect then re-enter it
-                if (flag == False):
+                # Loop that prompts if the PIN was incorrect then re-enter it
+                while (flag == False):
+                    array = None
+                    blinkKey2()
                     print("Enter PIN")
-                    # type in the user's PIN
-                    array = kp.KeyPadAuthor()
 
-                    # function to check if user exists and if PIN is correct
-                    flag = auth(ID, array, cursor)
+                    # interrupt handler that sets a timer in the background
+                    signal.signal(signal.SIGALRM, kp.timer)
+                    signal.alarm(7)
+
+                    try:
+                        # type in the user's PIN
+                        array = kp.KeyPadAuthor()
+
+                        # function to check if user exists and if PIN is correct
+                        flag = auth(ID, array, cursor)
```

Comment:
I'm concerned that this might ring up an obscene amount of SQL queries.

Would have to look into it much more - just something we should keep an eye on.